contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,779 @@
+     *            object will be created using whatever internal storage type the
+     *            creation function deems appropriate. Use the function
+     *            {@link #CFNumberGetType} to find out what type the
+     *            {@code CFNumber} object used to store your value.
+     * @param valuePtr
+     *            A pointer to the value for the returned number object.
+     * @return A new number with the value specified by {@code valuePtr}.
+     */
+    CFNumberRef CFNumberCreate(CFAllocatorRef alloc, long theType, PointerType valuePtr);
+
+    /**
+     * Creates a new immutable array with the given values.
+     * <p>
+     * This reference must be released with {@link #CFRelease} to avoid leaking
+     * references.
+     *
+     * @param alloc
+     *            The allocator to use to allocate memory for the new array and its
+     *            storage for values. Pass {@code null} or
+     *            {@code kCFAllocatorDefault} to use the current default allocator.
+     * @param values
+     *            A C array of the pointer-sized values to be in the new array. The
+     *            values in the new array are ordered in the same order in which
+     *            they appear in this C array. This value may be {@code null} if
+     *            {@code numValues} is 0. This C array is not changed or freed by
+     *            this function. If {@code values} is not a valid pointer to a C
+     *            array of at least {@code numValues} elements, the behavior is
+     *            undefined.
+     * @param numValues
+     *            The number of values to copy from the {@code values} C array into
+     *            the new array. This number will be the count of the new array—it
+     *            must not be negative or greater than the number of elements in
+     *            values.
+     * @param callBacks
+     *            A pointer to a {@code CFArrayCallBacks} structure initialized with
+     *            the callbacks for the array to use on each value in the
+     *            collection. The retain callback is used within this function, for
+     *            example, to retain all of the new values from the {@code values} C
+     *            array. A copy of the contents of the callbacks structure is made,
+     *            so that a pointer to a structure on the stack can be passed in or
+     *            can be reused for multiple collection creations.
+     *            <p>
+     *            This value may be {@code null}, which is treated as if a valid
+     *            structure of version 0 with all fields {@code null} had been
+     *            passed in.
+     * @return A new immutable array containing {@code numValues} from
+     *         {@code values}, or {@code null} if there was a problem creating the
+     *         object.
+     */
+    CFArrayRef CFArrayCreate(CFAllocatorRef alloc, Pointer values, long numValues, Pointer callBacks);
```

Comment:
I had the same thought about `Structure` but then realized it would be better to write the actual callback structures expected, complete with all the types inside the structure.  Which would then involve writing a unit test to test those structures and their callback functions.  

Which is something I still might do someday, but not today.